### DISCUSSION TITLE
can blend models be explained by shap?

### DISCUSSION THREAD

**Discussion by XiangBu** [pycaret/pycaret#1917] [Q&A] [answered]
As show in the following pic, I's like to blend lightgbt and lda to balance the AUC and recall, It turns out the result is good. However  it seems  blend models cannot be explained by shap. Is there any way to solve this?
![屏幕截图 2021-12-04 213423](https://user-images.githubusercontent.com/46986985/144711874-d1819219-4cc7-48a1-806a-dea732aa706f.png)

**Comment by moezali1** [comment, marked as answer]
Unfortunately not. Shap explainers work with only tree-based, kernel-based, or NN models. In PyCaret only tree-based models are supported for SHAP.

**Reply by XiangBu** [reply]
> Unfortunately not. Shap explainers work with only tree-based, kernel-based, or NN models. In PyCaret only tree-based models are supported for SHAP.

ok, thanks for your kind reply.

**Reply by XiangBu** [reply]
So I guess I could blend the model and use kernelexplainer if I don't use pycaret, right?

**Reply by Yard1** [reply]
You can also create a model with pycaret and pass it to kernelexplainer.

**Reply by Explosif3005** [reply]
Can we create a model on another library and take it to Pycaret a way or an another (because we can do the other thing as it seems...)

**Reply by Yard1** [reply]
You can pass any scikit learn compatible model to any pycaret function.

**Reply by Explosif3005** [reply]
ok thanks for the help!

**Reply by Explosif3005** [reply]
Just asking again what kind of library for AI is compatible with sckiti learn? Because some of them are interesting. Or do i have a way to do transfer learning with Pycaret or Skiti learn?
